pkg/p2p/libp2p/libp2p.go

```diff
@@ -187,12 +188,15 @@ func New(ctx context.Context, o Options) (*Service, error) {
 
 			s.logger.Debugf("handshake: handle %s: %v", peerID, err)
 			s.logger.Errorf("unable to handshake with peer %v", peerID)
-			// todo: test connection close and refactor
 			_ = s.disconnect(peerID)
 			return
 		}
 
-		s.peers.add(stream.Conn(), i.Address)
+		if s.peers.addIfNotExists(stream.Conn(), i.Address) {
```

Comment:
it would be nice if there is more explicit mentioning of the return value from the function, such as:
```go
if exists := s.peers.addIfNotExists(); exists {
  // peer already exists, don't notify again
  return
}
```